src/reanimated2/hook/useWorkletCallback.ts

```diff
@@ -1,10 +1,14 @@
 'use strict';
 import { useCallback } from 'react';
 import type { DependencyList } from './commonTypes';
+import type { WorkletFunction } from '../commonTypes';
 
 export function useWorkletCallback<Args extends unknown[], ReturnValue>(
   worklet: (...args: Args) => ReturnValue,
   deps?: DependencyList
 ) {
-  return useCallback(worklet, deps ?? []);
+  return useCallback(
+    worklet,
+    deps ?? Object.values((worklet as WorkletFunction).__closure)
```

Comment:
I'm wondering if this also works on web in case we don't use Reanimated Babel plugin.

Here's a code snippet from `useDerivedValue` that does exactly the same thing, maybe we can copy it over to here?
https://github.com/software-mansion/react-native-reanimated/blob/a2ae77e0386da38989eaebcd3ca1699ca00b6559/src/reanimated2/hook/useDerivedValue.ts#L22-L28